app/views/common/os_selection/_pxe_loader.html.erb

```diff
@@ -0,0 +1,12 @@
+<%= fields_for item do |f| -%>
+  <% loaders = @operatingsystem.available_loaders(item.architecture); if loaders.any? -%>
+    <% selected_attr = f.object.new_record? && item.hostgroup.present? && item.hostgroup.pxe_loader.nil? ? { :selected => @operatingsystem.preferred_loader(item.architecture) } : {} -%>
```

Comment:
I think this is different issue - when not using HG, then when architecture is set, the `item.architecture` is still nil, therefore correct intel suffix (32/64) is not generate. Only after you submit once (with error) architecture is now properly set.

You can see this when you click on the DHCP filename option help link.